x-pack/plugins/observability/public/pages/slo_outdated_definitions/outdated_slo_search_bar.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFlexGroup, EuiFlexItem, EuiButton, EuiFieldSearch } from '@elastic/eui';
+
+interface OutdatedSloSearchBarProps {
+  initialSearch?: string;
+  onRefresh: () => void;
+  onSearch: (search: string) => void;
+}
+
+export function OutdatedSloSearchBar({
+  onSearch,
+  onRefresh,
+  initialSearch = '',
+}: OutdatedSloSearchBarProps) {
+  const [tempSearch, setTempSearch] = useState<string>(initialSearch);
+  const [search, setSearch] = useState<string>(initialSearch);
+
+  const refreshOrUpdateSearch = () => {
+    if (tempSearch !== search) {
+      setSearch(() => {
+        onSearch(tempSearch);
+        return tempSearch;
+      });
+    } else {
+      onRefresh();
+    }
+  };
+
+  return (
+    <EuiFlexGroup gutterSize="s">
+      <EuiFlexItem>
+        <EuiFieldSearch
+          data-test-subj="o11ySlosOutdatedDefinitionsFieldSearch"
+          fullWidth
+          value={tempSearch}
+          onChange={(e) => setTempSearch(e.target.value)}
+        />
```

Comment:
May I suggest we trigger the search when enter is pressed? I did not find a callback on this component that does it without using `onKeyDown` but maybe some alternative ways exist.


```
onKeyDown={(event) => {
            if (event.key === 'Enter') {
              refreshOrUpdateSearch();
            }
          }}
```